yarn-project/p2p/src/testbench/p2p_client_testbench_worker.ts

```diff
@@ -0,0 +1,158 @@
+/**
+ * A testbench worker that creates a p2p client and listens for commands from the parent.
+ *
+ * Used when running testbench commands
+ */
+import { MockL2BlockSource } from '@aztec/archiver/test';
+import { P2PClientType, Tx, TxStatus, type WorldStateSynchronizer } from '@aztec/circuit-types';
+import { type EpochCacheInterface } from '@aztec/epoch-cache';
+import { EthAddress } from '@aztec/foundation/eth-address';
+import { createLogger } from '@aztec/foundation/log';
+import { sleep } from '@aztec/foundation/sleep';
+import { type DataStoreConfig } from '@aztec/kv-store/config';
+import { openTmpStore } from '@aztec/kv-store/lmdb-v2';
+
+import { type P2PConfig } from '../config.js';
+import { createP2PClient } from '../index.js';
+import { type AttestationPool } from '../mem_pools/attestation_pool/attestation_pool.js';
+import { type EpochProofQuotePool } from '../mem_pools/epoch_proof_quote_pool/epoch_proof_quote_pool.js';
+import { type TxPool } from '../mem_pools/tx_pool/index.js';
+import { AlwaysTrueCircuitVerifier } from '../test-helpers/reqresp-nodes.js';
+
+// Simple mock implementation
+function mockTxPool(): TxPool {
+  // Mock all methods
+  return {
+    addTxs: () => Promise.resolve(),
+    getTxByHash: () => Promise.resolve(undefined),
+    getArchivedTxByHash: () => Promise.resolve(undefined),
+    markAsMined: () => Promise.resolve(),
+    markMinedAsPending: () => Promise.resolve(),
+    deleteTxs: () => Promise.resolve(),
+    getAllTxs: () => Promise.resolve([]),
+    getAllTxHashes: () => Promise.resolve([]),
+    getPendingTxHashes: () => Promise.resolve([]),
+    getMinedTxHashes: () => Promise.resolve([]),
+    getTxStatus: () => Promise.resolve(TxStatus.PENDING),
+    getTxEffect: () => Promise.resolve(undefined),
+    getSettledTxReceipt: () => Promise.resolve(undefined),
+  } as TxPool;
```

Comment:
Does this work if we remove the cast? So we can get tsc alert us if we change a method in TxPool. Same for the other mocks.

Reply:
Ah it should!